pixy/httpapi.go

```diff
@@ -131,6 +133,18 @@ func (as *HTTPAPIServer) handleProduce(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if isSync {
+		switch err := as.kafkaClient.Produce(
+			topic, toEncoderPreservingNil(key), sarama.StringEncoder(message)); err {
+		case nil:
+			w.WriteHeader(http.StatusOK)
+		case sarama.ErrUnknownTopicOrPartition:
+			http.Error(w, err.Error(), http.StatusNotFound)
+		default:
+			http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
Will this ever get exposed to external users? If so we might want to log the exact error and return a vague one.